esp-hal/src/timer/systimer.rs

```diff
@@ -227,7 +227,7 @@ where
     DM: Mode,
 {
     /// Set the target value of this [Alarm]
-    pub fn set_target(&mut self, timestamp: u64) {
+    pub fn set_target(&self, timestamp: u64) {
```

Comment:
This was required to "fix" the embassy `SYSTIMER` time driver, and isn't really correct I guess. I guess `set_target` takes ticks while `load_value` take micros, and the conversion screws it up or something. I'm not really familiar with what's going on with this `Driver` trait honestly so not really sure what's going on, but it's working now at least.

Reply:
Oh, I guess because we read the current values of `hi` and `lo` inside of `load_value`, then add the ticks to that.

Reply:
Nice catch! Could we also change the other alarm methods to take `&self` so it's consistent? Maybe `set_interrupt_handler` should stay `&mut self` though, just to avoid possible race conditions when setting the handler? Not sure.